components/post/index.tsx

```diff
@@ -97,141 +97,144 @@ function removePostAndCloseRHS(post: ExtendedPost) {
+
+        const user = getUser(state, post.user_id);
+        const isBot = Boolean(user && user.is_bot);
+        const highlightedPostId = getHighlightedPostId(state);
+        const showActionsMenuPulsatingDot = showActionsDropdownPulsatingDot(state);
+        const tourTipsEnabled = onboardingTourTipsEnabled(state);
+        const selectedCard = getSelectedPostCard(state);
+
+        let emojis: Emoji[] = [];
+        const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true';
+        if (oneClickReactionsEnabled) {
+            emojis = getOneClickReactionEmojis(state);
+        }
 
-    const currentTeam = getCurrentTeam(state);
-    let teamName = currentTeam.name;
-    let teamDisplayName = '';
-
-    const memberships = getTeamMemberships(state);
-    const isDMorGM = channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL;
-    const rhsState = getRhsState(state);
-    if (
-        rhsState !== RHSStates.PIN && // Not show in pinned posts since they are all for the same channel
-        !isDMorGM && // Not show for DM or GMs since they don't belong to a team
-        memberships && Object.values(memberships).length > 1 // Not show if the user only belongs to one team
-    ) {
-        const team = getTeam(state, channel.team_id);
-        teamDisplayName = team?.display_name;
-        teamName = team?.name || currentTeam.name;
-    }
+        let previousPost = null;
+        if (ownProps.previousPostId) {
+            previousPost = getPost(state, ownProps.previousPostId);
+        }
 
-    const canReply = isDMorGM || (channel.team_id === currentTeam.id);
-    const directTeammate = getDirectTeammate(state, channel.id);
+        let previousPostIsComment = false;
 
-    const previewCollapsed = get(
-        state,
-        Preferences.CATEGORY_DISPLAY_SETTINGS,
-        Preferences.COLLAPSE_DISPLAY,
-        Preferences.COLLAPSE_DISPLAY_DEFAULT,
-    );
+        if (previousPost && !post.props.priority) {
+            previousPostIsComment = Boolean(previousPost.root_id);
+        }
 
-    const previewEnabled = getBool(
-        state,
-        Preferences.CATEGORY_DISPLAY_SETTINGS,
-        Preferences.LINK_PREVIEW_DISPLAY,
-        Preferences.LINK_PREVIEW_DISPLAY_DEFAULT === 'true',
-    );
+        const currentTeam = getCurrentTeam(state);
+        let teamName = currentTeam.name;
+        let teamDisplayName = '';
+
+        const memberships = getTeamMemberships(state);
+        const isDMorGM = channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL;
+        const rhsState = getRhsState(state);
+        if (
+            rhsState !== RHSStates.PIN && // Not show in pinned posts since they are all for the same channel
+            !isDMorGM && // Not show for DM or GMs since they don't belong to a team
```

Comment:
Variable name makes me think of [De Morgans' laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws), more so because its used in a bunch of chained logic statements :smile:

Reply:
Oh boy. It's been a while since I've thought of De Morgans' laws 😅